packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -248,6 +252,13 @@ export class MonacoWorkspace {
                 await this.performSnippetEdits(<MonacoResourceTextEdit[]>snippetEdits);
             }
 
+            // when enabled (option AND setting) loop over all dirty working copies and trigger save
+            // for those that were involved in this bulk edit operation.
+            const resources = new Set<string>(edits.filter(edit => edit instanceof MonacoResourceTextEdit).map(edit => (edit as MonacoResourceTextEdit).resource.toString()));
+            if (options?.respectAutoSaveConfig && this.editorPreferences.get('files.refactoring.autoSave') === true && resources.size > 0) {
+                this.saveAll(resources);
+            }
```

Comment:
Suggestion: Consequently, we should await here too.

```suggestion
           const resources = new Set<string>(
                edits
                    .filter((edit): edit is MonacoResourceTextEdit => edit instanceof MonacoResourceTextEdit)
                    .map(edit => edit.resource.toString())
            );
            if (resources.size > 0 && options?.respectAutoSaveConfig && this.editorPreferences.get('files.refactoring.autoSave') === true) {
                await this.saveAll(resources);
            }
```